geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala

```diff
@@ -0,0 +1,83 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.kafka.consumer
+
+import java.io.Closeable
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.{Executors, TimeUnit}
+
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.kafka.clients.consumer.{Consumer, ConsumerRecord}
+import org.apache.kafka.common.errors.WakeupException
+
+import scala.util.control.NonFatal
+
+trait ThreadedConsumer extends Closeable with LazyLogging {
+
+  protected val consumers: Seq[Consumer[Array[Byte], Array[Byte]]]
+  protected val topic: String
+  protected val frequency: Long
+
+  protected def consume(record: ConsumerRecord[Array[Byte], Array[Byte]]): Unit
+
+  private val closed = new AtomicBoolean(false)
+
+  private val executor = Executors.newFixedThreadPool(consumers.length)
```

Comment:
i almost want to have a dummy class for geomesa that extends this to make named threadpools so jstack instances show them easily.

Reply:
sure, add a ticket :)